### ISSUE TITLE
How to use keyStore?

### ISSUE THREAD

**Issue by Serobegi** [aws/amazon-freertos-ble-android-sdk#19]
```
esp32ListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                if(amazonFreeRTOSDevice != null){
                    amazonFreeRTOSManager.disconnectFromDevice(amazonFreeRTOSDevice);
                    amazonFreeRTOSDevice = null;
                }
                // esp32List.get(position).getBluetoothDevice().createBond() // 최후의 수단

                try {
                    amazonFreeRTOSDevice = amazonFreeRTOSManager
                            .connectToDevice(
                                    esp32List.get(position).getBluetoothDevice(),
                                    connectionStatusCallback,
                                    KeyStore.getInstance("String"),
                                    false);

                } catch (KeyStoreException e) {
                    e.printStackTrace();
                }
            }
        });
```

i don't know Keystore, help me.. plz

**Comment by Serobegi** [comment]
`2020-06-19 09:52:33.667 32636-32636/com.example.divide_scan_adver D/ViewRootImpl@d4b011d[FreeRTOS]: ViewPostIme pointer 0
2020-06-19 09:52:33.768 32636-32636/com.example.divide_scan_adver D/ViewRootImpl@d4b011d[FreeRTOS]: ViewPostIme pointer 1
2020-06-19 09:52:33.772 32636-32636/com.example.divide_scan_adver D/BluetoothAdapter: STATE_ON
2020-06-19 09:52:33.774 32636-32636/com.example.divide_scan_adver I/AmazonFreeRTOSManager: Starting ble device scan
2020-06-19 09:52:33.775 32636-32636/com.example.divide_scan_adver D/BluetoothAdapter: STATE_ON
2020-06-19 09:52:33.777 32636-32636/com.example.divide_scan_adver D/BluetoothAdapter: STATE_ON
2020-06-19 09:52:33.777 32636-32636/com.example.divide_scan_adver D/BluetoothLeScanner: Start Scan with callback
2020-06-19 09:52:33.781 32636-32701/com.example.divide_scan_adver D/BluetoothLeScanner: onScannerRegistered() - status=0 scannerId=6 mScannerId=0
2020-06-19 09:52:53.796 32636-1348/com.example.divide_scan_adver I/AmazonFreeRTOSManager: Stopping ble device scan
2020-06-19 09:52:53.804 32636-1348/com.example.divide_scan_adver D/BluetoothAdapter: STATE_ON
2020-06-19 09:52:53.805 32636-1348/com.example.divide_scan_adver D/BluetoothLeScanner: Stop Scan with callback`

**Comment by ravibhagavandas** [comment]
@Serobegi 

What are you using the keystore for ?  Keystore option provided in the demo app is used for storing the certificate and private key used for connecting to AWS IoT MQTT broker.  Is that your intended usage ?

If so you can look into [AWSIoTKeyStoreHelper ](https://aws-amplify.github.io/aws-sdk-android/docs/reference/com/amazonaws/mobileconnectors/iot/AWSIotKeystoreHelper.html) class on how to generate a key pair in keystore and pass the keystore to the SDK.

For security purpose we recommend to use a cognito based authentication on the mobile devices, as it does not involve storing a keypair on the mobile device. You can find more documentation on how to set up cognito for the demo [here](https://docs.aws.amazon.com/freertos/latest/userguide/ble-demo.html#ble-sdk-app).

**Comment by Serobegi** [comment]
Hello!

I don't need MQTT function, I am making an app that can make BLE connection with the board and send and receive WIFI information only.

Can you tell me how to use WSIoTKeyStoreHelper class?

And thank you so much for replying.

**Comment by ravibhagavandas** [comment]
`AWSCredentialsProvider awsCredentialsProvider = AWSMobileClient.getInstance();` This uses Cognito based authentication (which is the recommended way for user authentication on mobile app) instead of keystore.

You can find a sample of how to create kes and load them from keystore using AwsIoTkeystoreHelper class here: https://github.com/awslabs/aws-sdk-android-samples/blob/main/AndroidPubSub/src/com/amazonaws/demo/androidpubsub/PubSubActivity.java#L203

The sample creates a key pair and certificate using `createKeysAndCertificate()` API and saves it to the keystore.

**Comment by Darsh-Dev** [comment]
hi @ravibhagavandas ,

> You can find a sample of how to create keys and load them from keystore using AwsIoTkeystoreHelper class here: https://github.com/awslabs/aws-sdk-android-samples/blob/main/AndroidPubSub/src/com/amazonaws/demo/androidpubsub/PubSubActivity.java#L203
I can't open this link (page not found).

I would like to use the keystore instead of congnito based solution. Can you please share the steps/document?

**Comment by ravibhagavandas** [comment]
For security concerns we now recommend to always use Cognito instead of creating certificates/private key for mobile apps. All of our current examples recommend using Cognito as the means to authenticate a websocket connection to AWS IoT. You can get more details on Cognito authentication from the [demo](https://github.com/aws/amazon-freertos-ble-android-sdk/blob/master/app/src/main/java/software/amazon/freertos/demo/AuthenticatorActivity.java) in this repository.

Please note the Keystore helper class and related files have been moved under aws_amplify repository [here](https://github.com/aws-amplify/aws-sdk-android/blob/main/aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotKeystoreHelper.java)

**Comment by Darsh-Dev** [comment]
@ravibhagavandas 
I would like to check the Device Certificate blacklist, whitelist, revoke, OCSP stapling feature of BLE end node device. Can we achieve similar feature using Cognito based solution?

**Comment by ravibhagavandas** [comment]
Could you elaborate more on your use case? By BLE end node device does it mean an end micro-controller device here owns the certificate? If so I am not sure how a certificate on mobile app  can helps this. Cognito uses the user name and password based authentication to create short lived tokens (more details [here](https://docs.aws.amazon.com/cognito/latest/developerguide/what-is-amazon-cognito.html)) hence you don't need to handle certificates and blacklisting/revocation etc.

**Comment by Darsh-Dev** [comment]
We have used the Optiga TrustM HSM on BLE based end device. We have added Mutual authentication and data encrypt/decrypt functionality and its working. Now, we would like to share the device certification with AWS cloud and needs to validate Device Certificate blacklist, whitelist, revoke, OCSP stapling features. 

I want to understand, how Cognito is more secured then certification based model? How we can blacklist/revoke the device from AWS cloud?
